src/main/java/org/primefaces/component/avatar/AvatarRenderer.java

```diff
@@ -92,4 +106,49 @@ public void encodeChildren(FacesContext context, UIComponent component) throws I
     public boolean getRendersChildren() {
         return true;
     }
+
+    /**
+     * Generate a Gravatar URL for an email addressed based on API docs.
+     *
+     * @see https://en.gravatar.com/site/implement/images/
+     * @param context the Face
+     * @param avatar the Avatar to create a Gravatar for
+     * @return the URL to retrieve the Gravatar image
+     */
+    protected String generateGravatar(FacesContext context, Avatar avatar) {
+        String email = avatar.getGravatar();
+        String config = avatar.getGravatarConfig();
+        String url = null;
+        try {
+            StringBuilder sb = SharedStringBuilder.get(context, SB_AVATAR);
+            sb.append(GRAVATAR_URL);
+            sb.append(generateMailHash(email));
+            if (LangUtils.isNotBlank(config)) {
+                sb.append('?').append(config);
+            }
+            url = sb.toString();
+        }
+        catch (NoSuchAlgorithmException e) {
+            throw new FacesException("Failed to generate Gravatar URL for value: " + email);
+        }
+        return url.toString();
+    }
+
+    /**
+     * Converts email address into a hash code for Gravatar API.
+     *
+     * @param email the email to encode
+     * @return a MD5 hashcode of the email address
+     * @throws NoSuchAlgorithmException if hash can't be encoded
+     */
+    protected String generateMailHash(String email) throws NoSuchAlgorithmException {
```

Comment:
why not pass the existing StringBuilder here?

Reply:
Great idea. Done!